arangod/Agency/Node.h

```diff
@@ -286,26 +176,13 @@ class Node final {
   /// @brief Is object
   bool isObject() const;
 
-  /**
-   * @brief Set expiry for this node
-   * @param Time point of expiry
-   */
-  void timeToLive(TimePoint const& ttl);
-
   /// @brief accessor to Node object
   /// @return  returns nullptr if not found or type doesn't match
-  Node const* hasAsNode(std::string const&) const noexcept;
-
-  /// @brief accessor to Node object
-  Node& hasAsWritableNode(std::string const&);
-
-  /// @brief accessor to Node's type
-  /// @return  returns nullopt if not found or type doesn't match
-  std::optional<NodeType> hasAsType(std::string const&) const noexcept;
+  std::shared_ptr<Node const> hasAsNode(std::string const&) const noexcept;
 
   /// @brief accessor to Node's Slice value
   /// @return  returns nullopt if not found or type doesn't match
-  std::optional<velocypack::Slice> hasAsSlice(
+  std::optional<velocypack::String> hasAsSlice(
```

Comment:
This has been and is dangerous and is used only a few times in the code base. Let's remove it and replace calls with other methods.